collect/collect.go

```diff
@@ -228,6 +245,76 @@ func (i *InMemCollector) checkAlloc() {
 	runtime.GC()
 }
 
+func (i *InMemCollector) newCheckAlloc() {
+	inMemConfig, err := i.Config.GetInMemCollectorCacheCapacity()
+
+	var mem runtime.MemStats
+	runtime.ReadMemStats(&mem)
+	i.Metrics.Gauge("memory_heap_allocation", int64(mem.Alloc))
+	if err != nil || inMemConfig.MaxAlloc == 0 || mem.Alloc < inMemConfig.MaxAlloc {
+		return
+	}
+
+	// Figure out what fraction of the total cache we should remove. We'd like it to be
+	// enough to get us below the max capacity, but not TOO much below.
+	// Because our impact numbers are only the data size, we shoot for 90% of max
+	// capacity and we should end up comfortably below that.
+	totalToRemove := mem.Alloc - inMemConfig.MaxAlloc //*9/10
```

Comment:
What is `//*9/10` implying?

Reply:
Heh, that I had a 90% in there but decided it was too dependent on cache details so removed it. I've cleaned up the comment.